internal/component/otelcol/config_http.go

```diff
@@ -21,13 +21,9 @@ type HTTPServerArguments struct {
 
 	CORS *CORSArguments `alloy:"cors,block,optional"`
 
-	// TODO(rfratto): auth
-	//
-	// Figuring out how to do authentication isn't very straightforward here. The
-	// auth section links to an authenticator extension.
-	//
-	// We will need to generally figure out how we want to provide common
-	// authentication extensions to all of our components.
+	// Auth is a binding to an otelcol.auth.* component extension which handles
+	// authentication.
+	Auth *auth.Handler `alloy:"auth,attr,optional"`
```

Comment:
I think we should name this to a `authentication` in both the var name and alloy name. This is so it doesnt overlap with `authentication`. Same with the documentation.

Reply:
What do you think about renaming it to Authentication internally but keeping the alloy export as auth? That way when using client auth or server auth the field is consistent?

client auth looks like this. We can't easily change this without breaking the user interface.
```
otelcol.exporter.otlp "otlp"{
  auth = <component>.handler
}

server auth would look like this if we changed auth to authentication. 
```
otelcol.receiver.otlp "receiver"{
    authentication = <auth-component>.handler
}
```

As a user I like the consistency between it being auth for both exporters and receivers. Let me know what you think.

Reply:
I can live with that. Can you add a comment in the code saying why its that way?